### ISSUE TITLE
NoLineWrap rule : false-positive on annotations with CLASS_DEF,METHOD_DEF (6.12.1)

### ISSUE THREAD

**Issue by lautou** [checkstyle/checkstyle#2833]
TestClass.java

    package test;
    
    @MyClassAnnotation
    public class TestClass {
    
        @MyMethodAnnotation
        String method() {
        }
    }

config.xml

    <?xml version="1.0"?>
    <!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
   
       <module name="TreeWalker">
            <module name="NoLineWrap"/>
        </module>

We get false-positive defects on annotations when using NoLineWrap rule with CLASS_DEF,METHOD_DEF 

This issue was detected using CheckStyle 6.12.1.

**Comment by romani** [comment]
thanks for reporting, please change description to conform with http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug, it wil help quickly reproduce issue and fix it.

**Comment by romani** [comment]
can not reproduce on latest version.

```
$ cat Test.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <module name="TreeWalker">
        <module name="NoLineWrap"/>
    </module>
</module>

$ cat TestClass.java 
@MyClassAnnotation
public class TestClass {

    @MyMethodAnnotation
    String method() {
    }
}

$ java -jar checkstyle-6.14.1-all.jar -c Test.xml Main.java 
Starting audit...
Audit done.
```

**Comment by gmsdelmundo** [comment]
@romani Were you running checkstyle against the right class? I saw your command using `Main.java` instead of `TestClass.java`.

I can reproduce it with the following:

```
$ cat checkstyle.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <module name="TreeWalker">
        <module name="NoLineWrap"/>
    </module>
</module>
```

```
$ cat TestClass.java
@MyClassAnnotation
public class TestClass {
    @MyMethodAnnotation
    String method() {
    }
}
```

```
$ java -jar checkstyle-10.23.0-all.jar -c checkstyle.xml TestClass.java 
Starting audit...
[ERROR] C:\path\to\TestClass.java:3:1: METHOD_DEF statement should not be line-wrapped. [NoLineWrap]
Audit done.
[ERROR] C:\path\to\TestClass.java:5:5: METHOD_DEF statement should not be line-wrapped. [NoLineWrap]
Audit done.
Checkstyle ends with 1 error
```

**Comment by romani** [comment]
@gmsdelmundo , please create new issue.
